### DISCUSSION TITLE
Training Steps

### DISCUSSION THREAD

**Discussion by karzanclick** [fishaudio/fish-speech#564] [Q&A] [answered]
First, I'd like to express my gratitude for the excellent fish-speech model. It's truly impressive work.
I'm looking to train this model for new language, but I'm uncertain about the optimal number of training steps to use.

My dataset details:

76,000 voice samples
Approximately 74 hours of audio
Around 20 different speakers

Given these dataset characteristics, could you recommend an appropriate number of training steps? Any advice on optimizing the training process would be greatly appreciated.
Thank you again for developing such a valuable tool for the community.

**Comment by Maharshi222** [comment, marked as answer]
With 76,000 samples, a batch size of 32 gives you 2,375 batches per epoch.
A good starting point for speech models is to aim for around 10–20 epochs.
Therefore, total steps = epochs × steps per epoch.

10 epochs: 23,750 steps.
20 epochs: 47,500 steps.

**Reply by karzanclick** [reply]
Thank you for your recommendation! I'll give it a try and provide feedback on the results after testing

**Reply by abhisirka2001** [reply]
Hey @karzanclick 
Did you get good results?
Can you help me finetuning the model for multiple languages?

**Reply by cod3r0k** [reply]
Do you have any update?

**Comment by dluebke** [comment]
Hi, I just found this thread and I am also interested in training a custom fish speech model. However, I could not find any information about how to do this. Could you please provide me a pointer to some documentation? Thanks, Daniel

**Reply by Ain477** [reply]
seems like no one is going to explain this.
I'm also struggling with this
